src/stepFunctions/commands/visualizeStateMachine/getStateMachineDefinitionFromCfnTemplate.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs'
+import { getLogger, Logger } from '../../../shared/logger'
+
+/**
+ * @param {string} uniqueIdentifier - unique identifier of state machine
+ * @param {string} templatePath - path for the template.json file
+ *
+ * @returns the escaped ASL Json definition string of the state machine construct
+ */
+export function getStateMachineDefinitionFromCfnTemplate(
+    uniqueIdentifier: string,
+    templatePath: string
+): string | undefined {
+    const logger: Logger = getLogger()
+    try {
+        const data = fs.readFileSync(templatePath, 'utf8')
+        const jsonObj = JSON.parse(data)
+        const resources = jsonObj.Resources
+        let key: string
+
+        const matchingKeyList: string[] = []
+        for (const key of Object.keys(resources)) {
+            //the resources list always contains 'CDKMetadata'
+            if (key === 'CDKMetadata') {
+                continue
+            }
+
+            if (key.substring(0, uniqueIdentifier.length) === uniqueIdentifier) {
+                matchingKeyList.push(key)
+            }
+        }
+        if (matchingKeyList.length === 0) {
+            return
+        } else if (matchingKeyList.length === 1) {
```

Comment:
Unnecessary `else if`, the `reduce` statement in the `else` should work for this.